File: FizzBuzz/ruby/fizzbuzz.rb
def fbprint x
  if x % 15 == 0
    return "FizzBuzz"
  elsif x % 3 == 0
    return "Fizz"
  elsif x % 5 == 0
    return "Buzz"
  end

  x
end

if __FILE__ == $0
  (1..100).each { |x| puts(fbprint x) }
end

File: FizzBuzz/scala/fizzbuzz.scala
#!/usr/bin/env scala

def fbprint(x:Int) = x match {
  case i:Int if i % 15 == 0 => "FizzBuzz"
  case i:Int if i %  3 == 0 => "Fizz"
  case i:Int if i %  5 == 0 => "Buzz"
  case _                    => x
}

(1 until 100).foreach { x =>
  println(fbprint(x))
}
